### PR TITLE
Feat/role change request

### PR DESCRIPTION
**Pull request by ashwins93**
## Details

- Added user settings button against each participant in the list
- New user settings dialog with a role change dropdown
- Detect change in role settings and call change role from store.

### Choose one of these(put a 'x' in the bracket):
- [ ] The change doesn't require a change to the documentation.
- [ ] The documentation is updated accordingly.

### CODE REVIEW

File: src/components/ParticipantList/ParticipantInList.tsx [hunk 1]
```diff
@@ -26,6 +31,19 @@ export const ParticipantInList = ({
         </Text>
       </div>
       <div className={styler('menuIconContainer')}>
+        {!isLocal && (
+          <div>
+            <Button
+              iconOnly
+              variant={'no-fill'}
```
Comment (raviteja83) [inline src/components/ParticipantList/ParticipantInList.tsx:38]: could you use variant="no-fill". No need of braces for string. we will a lint rule later. same for other fields

File: src/components/ParticipantList/ParticipantList.tsx [hunk 2]
```diff
@@ -73,67 +107,167 @@ export const ParticipantList = ({
...
+          {listOpen && (
+            <div
+              className={`${styler('menuRoot')}`}
+              role="menu"
+              aria-orientation="vertical"
+              aria-labelledby="menu-button"
+              tabIndex={-1}
+            >
+              {roles &&
+                //@ts-expect-error
```
Comment (raviteja83) [inline src/components/ParticipantList/ParticipantList.tsx:238]: can we avoid this?

**Review by raviteja83 — COMMENTED** [review]
Hey, could you also add a screenshot.

**Comment by ashwins93** [comment]
Screenshots

screen1
screen2

File: src/components/ParticipantList/ParticipantInList.tsx (hunk 1 above)
Comment (ashwins93) [reply src/components/ParticipantList/ParticipantInList.tsx]: done

File: src/components/ParticipantList/ParticipantInList.tsx [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
 import { Text } from '../Text';
 import { ParticipantListClasses } from './ParticipantProps';
 import { StylerType } from '../../types';
+import { HMSPeer } from '@100mslive/hms-video-store';
 
 interface PropsType {
   styler?: StylerType<ParticipantListClasses>;
   name: string;
   isAudioEnabled?: boolean;
+  onRoleChangeClick?: () => void;
```
[the remark this answers is not in the document]
Comment (ashwins93) [reply src/components/ParticipantList/ParticipantInList.tsx]: good point, actually this isn't specific to role change, changed the name to `onUserSettingsClick`

File: src/components/ParticipantList/ParticipantList.tsx (hunk 2 above)
Comment (ashwins93) [reply src/components/ParticipantList/ParticipantList.tsx]: I have simplified the types.

**Review by triptu — APPROVED** [review]

**Pull request merged**